### PR TITLE
Cursor settings

### PR DESCRIPTION
**Pull request by devvoid**
# Summary of the PR
Add cursor settings to the GLFW input backend.

# Related issues, Discord discussions, or proposals
None.

# Further Comments
None.

### CODE REVIEW

**Comment by Pyrdacor** [comment]
Looking forward to this. The cursor show/hide feature is essential for my project and most games.

**Comment by devvoid** [comment]
I hope to have this finished soon! Christmas stuff has kept me pretty busy, but I'll have time to finish things up soon.

File: src/Input/Silk.NET.Input.Common/Interfaces/IMouse.cs [hunk 1]
```diff
@@ -1,60 +1,66 @@
...
+        
+        /// <summary>
+        /// The position of the cursor.
+        /// </summary>
+        PointF Position { get; set; }
+        
+        /// <summary>
+        /// The cursor to use.
+        /// </summary>
+        ICursor Cursor { get; set; }
```
Comment (Perksey) [inline src/Input/Silk.NET.Input.Common/Interfaces/IMouse.cs]: I don't think we should make this a setter as it makes implementation hell. Instead, we should ensure that all of ICursor's members are writeable.

File: src/Input/Silk.NET.Input.Common/Interfaces/ICursor.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+{
+    /// <summary>
+    /// An interface representing a mouse cursor.
+    /// </summary>
+    public interface ICursor
+    {
+        /// <summary>
+        /// Hotspot on the X axis.
+        /// </summary>
+        int HotspotX { get; set; }
```
Comment (Perksey) [inline src/Input/Silk.NET.Input.Common/Interfaces/ICursor.cs]: Is this not just `IMouse.X`?

Comment (devvoid) [reply src/Input/Silk.NET.Input.Common/Interfaces/ICursor.cs]: X and Y would be the cursor's position on-screen, not the hotspot, which is where the click actually happens.

**Comment by Pyrdacor** [comment]
How is the state?

**Comment by Perksey** [comment]
@Pyrdacor This hasn't moved along much, mainly because I recently refactored the entire GLFW input backend, but also because Void is pretty busy atm.

File: src/Input/Silk.NET.Input.Common/Interfaces/ICursor.cs (hunk 2 above)
Comment (Pyrdacor) [reply src/Input/Silk.NET.Input.Common/Interfaces/ICursor.cs]: Is the hotspot relative to the cursor position, e.g. (0,4)?

Comment (devvoid) [reply src/Input/Silk.NET.Input.Common/Interfaces/ICursor.cs]: Yeah

**Comment by devvoid** [comment]
I unfortunately don't have that much time to work on Silk right now. If you're able to take a crack at it, it'd be very much appreciated!

**Comment by Pyrdacor** [comment]
Ok I implemented a bunch of stuff in #108. Maybe you can have a look at it.

**Comment by Perksey** [comment]
#108 definitely is very close and you're being a machine with your PRs!!!! Thanks a ton :DDD

**Comment by Perksey** [comment]
Will merge once CI has passed

**Pull request merged**